pkg/cli/common.go

```diff
@@ -261,5 +261,10 @@ func GetDefaultAuthFile() string {
 	if runtimeDir != "" {
 		return filepath.Join(runtimeDir, "containers/auth.json")
 	}
-	return ""
+
+	uid := os.Getuid()
+	if uid == 0 {
+		return fmt.Sprintf("/run/containers/%d/auth.json", os.Getuid())
+	}
+	return fmt.Sprintf("/run/user/%d/containers/auth.json", os.Getuid())
```

Comment:
This differs from https://github.com/containers/libpod/pull/3752 and I'm not sure it's quite right as I think root would hit 267 when it should hit 269.

Reply:
The actual behavior depends on whether `XDG_RUNTIME_DIR` is set, and su/sudo clears it. It is not strictly specific to UID=0.